packages/app/src/client/views/ring-health/RingTable.tsx

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2021 Opstrace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useState } from "react";
+
+import axios from "axios";
+import {
+  TableContainer,
+  Table,
+  TableHead,
+  TableRow,
+  TableCell,
+  TableBody
+} from "@material-ui/core";
+import { Card } from "client/components/Card";
+import { format } from "date-fns";
+
+type Shard = {
+  id: string;
+  state: string;
+  timestamp: string;
+  zone: string;
+  address: string;
+  tokens: number[];
+  registered_timestamp: string;
+};
+
+type Payload = {
+  shards: Array<Shard>;
+  now: Date;
+};
+
+type Props = {
+  ringEndpoint: string;
+};
+
+const RingTable = ({ ringEndpoint }: Props) => {
+  const [shards, setShards] = useState<Array<Shard>>();
+  useEffect(() => {
+    async function fetchShards() {
+      const response = await axios.get<Payload>(ringEndpoint);
+      setShards(response.data.shards);
```

Comment:
I need to add error handling. Do we already have error boundaries setup somewhere? How should errors be handled?

Reply:
We do have a top-level ErrorBoundary but in this case, let's just use https://material-ui.com/components/alert/#alert and show a red Alert in place of the table, with the error as the message